tfhe/src/core_crypto/commons/math/random/generator.rs

```diff
@@ -184,6 +186,40 @@ impl<G: ByteRandomGenerator> RandomGenerator<G> {
         Scalar::fill_slice(self, Uniform, output);
     }
 
+    /// Fill a slice with random uniform values, for non-native power of 2 moduli, a shift is
+    /// applied to only keep log2(modulus) MSBs and zeroed out LSBs
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use concrete_csprng::generators::SoftwareRandomGenerator;
+    /// use concrete_csprng::seeders::Seed;
+    /// use tfhe::core_crypto::commons::math::random::RandomGenerator;
+    /// use tfhe::core_crypto::commons::parameters::CiphertextModulus;
+    /// let mut generator = RandomGenerator::<SoftwareRandomGenerator>::new(Seed(0));
+    /// let mut vec = vec![1u32; 100];
+    /// generator.fill_slice_with_random_uniform_custom_mod(
+    ///     &mut vec,
+    ///     CiphertextModulus::try_new_power_of_2(31).unwrap(),
+    /// );
+    /// ```
+    pub fn fill_slice_with_random_uniform_custom_mod<Scalar>(
+        &mut self,
+        output: &mut [Scalar],
+        ciphertext_modulus: CiphertextModulus<Scalar>,
+    ) where
+        Scalar: UnsignedInteger + RandomGenerable<Uniform>,
+    {
+        self.fill_slice_with_random_uniform(output);
+
+        if !ciphertext_modulus.is_native_modulus() {
+            slice_wrapping_scalar_mul_assign(
+                output,
+                ciphertext_modulus.get_scaling_to_native_torus(),
+            );
+        }
+    }
```

Comment:
@jborfila just need your opinion on this function to generate random uniform, I think it looks ok.

Random gaussian noise does not have scaling managed at the moment (i.e. it scales with the native type and not the user provided modulus)